Src/Core/Collections/GenericDictionaryAssertions.cs

```diff
@@ -551,6 +551,86 @@ public GenericDictionaryAssertions(IDictionary<TKey, TValue> dictionary)
         /// Asserts that the current dictionary contains the specified <paramref name="expected"/>.
         /// Keys and values are compared using their <see cref="object.Equals(object)" /> implementation.
         /// </summary>
+        /// <param name="expected">The expected key/value pairs.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="reasonArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<GenericDictionaryAssertions<TKey, TValue>> Contain(IEnumerable<KeyValuePair<TKey, TValue>> expected,
+            string because = "", params object[] reasonArgs)
+        {
+            if (expected == null)
+            {
+                throw new ArgumentNullException("expected", "Cannot compare dictionary with <null>.");
+            }
+
+            KeyValuePair<TKey, TValue>[] expectedKeyValuePairs = expected.ToArray();
+
+            if (!expectedKeyValuePairs.Any())
+            {
+                throw new ArgumentException("Cannot verify key containment against an empty dictionary");
+            }
+
+            if (ReferenceEquals(Subject, null))
+            {
+                Execute.Assertion
+                    .BecauseOf(because, reasonArgs)
+                    .FailWith("Expected {context:dictionary} to contain key/value pairs {0}{reason}, but found {1}.", expected, Subject);
+            }
+
+            var expectedKeys = expectedKeyValuePairs.Select(keyValuePair => keyValuePair.Key).ToArray();
+            var missingKeys = expectedKeys.Where(key => !Subject.ContainsKey(key));
+
+            if (missingKeys.Any())
+            {
+                if (expectedKeyValuePairs.Count() > 1)
```

Comment:
If you want, you can do some pretty advanced chaining to keep your code smaller. See https://github.com/dennisdoomen/fluentassertions/blob/master/Src/Core/Collections/CollectionAssertions.cs#L1169